client/state/sites/seo/test/reducer.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+
+/**
+ * Internal dependencies
+ */
+import { SEO_TITLE_SET } from 'state/action-types';
+
+import { titleFormats } from '../reducer';
+
+const siteA = 1;
+const siteB = 2;
+
+const siteNameTagline = '%site_name% | %tagline%';
+const postTitleSiteName = '%post_title% | %site_name%';
+
+describe( 'reducer', () => {
+	describe( '#titleFormats()', () => {
+		const prev = {
+			[ siteA ]: {
+				frontPage: siteNameTagline
+			}
+		};
+
+		it( 'should return empty structure when initializing', () => {
+			expect( titleFormats( undefined, {} ) ).to.eql( {} );
+		} );
+
+		it( 'should add new formats to an empty structure', () => {
+			expect( titleFormats( undefined, {
+				type: SEO_TITLE_SET,
+				siteId: siteA,
+				pageType: 'frontPage',
+				format: siteNameTagline
+			} ) ).to.eql( {
+				[ siteA ]: {
+					frontPage: siteNameTagline
+				}
+			} )
+		} );
+
+		it( 'should add new formats to an existing structure', () => {
+			expect( titleFormats( prev, {
```

Comment:
Let's `deepFreeze` the initial state so we can detect when we have accidental mutations